packages/oauth/src/providers/linkedin.ts

```diff
@@ -47,11 +47,10 @@ export const linkedin = <_Auth extends Auth>(auth: _Auth, config: Config) => {
 
 	const getProviderUser = async (accessToken: string) => {
 		const linkedinProfile = await getProfile(accessToken);
-		const displayImageElement = linkedinProfile.profilePicture[
-			"displayImage~"
-		]?.elements
-			?.slice(-1)
-			?.pop();
+		const displayImageElement = linkedinProfile.profilePicture && 
```

Comment:
I'm not a big fan of using `&&`. I think this is better?

```ts
linkedinProfile?.profilePicture["displayImage~"]?.elements
            ?.slice(-1)
            ?.pop() ?? null;
```

Reply:
the problem is that this will not solve the issue. The issue happens because of `profilePicture["displayImage~"]` . This is trying to get `displayImage~`from `profilePicture` while profile picture might not exist. The code you've suggested won't solve the issue as it will only test for `linkedinProfile` and not the `profilePicture` itself. We could solve it using a shorthand if but I thought the `&&`would be better.

Reply:
Would this work?

```ts
linkedinProfile?.profilePicture?.["displayImage~"]?.elements
            ?.slice(-1)
            ?.pop() ?? null;
```

Reply:
Yes it workds